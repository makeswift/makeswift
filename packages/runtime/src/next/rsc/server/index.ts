import 'server-only'

export { NextRSCMakeswiftPage as ExperimentalMakeswiftPage } from './makeswift-page'
export { NextRSCServerProvider as ExperimentalServerProvider } from './server-provider'
