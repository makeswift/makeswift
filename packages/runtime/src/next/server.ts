export { MakeswiftApiHandler } from './api-handler'
export type { Manifest, Font } from '../api-handler'
export type { MakeswiftApiHandlerResponse } from './api-handler'
export { getSiteVersion } from './draft-mode'
