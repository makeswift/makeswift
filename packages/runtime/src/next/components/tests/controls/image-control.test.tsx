/** @jest-environment jsdom */

import { Image } from '@makeswift/controls'
import { testPageControlPropRendering } from './page-control-prop-rendering'
import { http, HttpResponse } from 'msw'
import { server } from '../../../../mocks/server'
import { APIResourceType } from '../../../../api'

const imageId = 'image-test-id'

const filesBaseUrl = `/api/makeswift/files`

const fileResource = {
  __typename: APIResourceType.File,
  id: imageId,
  name: 'image',
  extension: 'jpg',
  publicUrl: 'https://example.com/image.jpg',
  dimensions: {
    width: 100,
    height: 100,
  },
}

describe('Page', () => {
  describe('Image control data v0', () => {
    describe.each([Image.Format.URL, Image.Format.WithDimensions, undefined])(
      `with format: %s`,
      format => {
        test(`fetching image`, async () => {
          server.use(http.get(`${filesBaseUrl}/${imageId}`, () => HttpResponse.json(fileResource)))
          await testPageControlPropRendering(Image({ format }), {
            value: {
              type: 'makeswift-file',
              id: imageId,
            },
            expectedRenders: 2,
            toData: val => ('id' in val ? val.id : 'undefined'),
          })
        })

        test(`reading file from cache`, async () => {
          server.use(http.get(`${filesBaseUrl}/${imageId}`, () => HttpResponse.json(fileResource)))
          await testPageControlPropRendering(Image({ format }), {
            value: {
              type: 'makeswift-file',
              id: imageId,
            },
            expectedRenders: 1,
            cacheData: {
              apiResources: {
                File: [{ id: imageId, value: fileResource }],
              },
            },
            toData: val => ('id' in val ? val.id : 'undefined'),
          })
        })
      },
    )
  })

  describe('ImageDefinition v1 reading v0 data', () => {
    describe.each([Image.Format.URL, Image.Format.WithDimensions, undefined])(
      `with format: %s`,
      format => {
        test(`fetching image`, async () => {
          server.use(http.get(`${filesBaseUrl}/${imageId}`, () => HttpResponse.json(fileResource)))
          await testPageControlPropRendering(Image({ format }), {
            value: {
              type: 'makeswift-file',
              id: imageId,
            },
            expectedRenders: 2,
          })
        })

        test(`reading file from cache`, async () => {
          server.use(http.get(`${filesBaseUrl}/${imageId}`, () => HttpResponse.json(fileResource)))
          await testPageControlPropRendering(Image({ format }), {
            value: {
              type: 'makeswift-file',
              id: imageId,
            },
            expectedRenders: 1,
            cacheData: {
              apiResources: {
                File: [{ id: imageId, value: fileResource }],
              },
            },
          })
        })
      },
    )
  })

  describe('Image control data v1', () => {
    describe.each([
      {
        value: {
          type: 'makeswift-file',
          id: imageId,
        } as const,
      },
    ])(`with value: $value`, ({ value }) => {
      describe.each([Image.Format.URL, Image.Format.WithDimensions, undefined])(
        `with format: %s`,
        format => {
          test(`fetching image`, async () => {
            server.use(
              http.get(`${filesBaseUrl}/${value.id}`, () => HttpResponse.json(fileResource)),
            )
            await testPageControlPropRendering(Image({ format }), {
              value,
              expectedRenders: 2,
            })
          })

          test(`reading file from cache`, async () => {
            server.use(
              http.get(`${filesBaseUrl}/${value.id}`, () => HttpResponse.json(fileResource)),
            )
            await testPageControlPropRendering(Image({ format }), {
              value,
              expectedRenders: 1,
              cacheData: {
                apiResources: {
                  File: [{ id: value.id, value: fileResource }],
                },
              },
            })
          })
        },
      )
    })
  })

  describe('Image control data v1 (external image)', () => {
    describe.each([
      {
        value: {
          type: 'external-file',
          url: 'https://example.com/image.jpg',
          width: 100,
          height: 100,
        } as const,
      },
    ])(`with value: $value`, ({ value }) => {
      describe.each([Image.Format.URL, Image.Format.WithDimensions, undefined])(
        `with format: %s`,
        format => {
          test(`rendering external image`, async () => {
            await testPageControlPropRendering(Image({ format }), {
              value,
              expectedRenders: 1,
            })
          })
        },
      )
    })
  })
})
