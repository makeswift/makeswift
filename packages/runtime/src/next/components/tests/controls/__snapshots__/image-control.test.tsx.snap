// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Page Image control data v0 with format: makeswift::controls::image::format::url fetching image: resolvedValue 1`] = `"https://example.com/image.jpg"`;

exports[`Page Image control data v0 with format: makeswift::controls::image::format::url fetching image: snapshot 1`] = `
{
  "cacheData": {
    "apiResources": {},
    "localizedResourcesMap": {},
  },
  "document": {
    "data": {
      "key": "00000000-0000-0000-0000-000000000000",
      "props": {
        "children": {
          "columns": [
            {
              "deviceId": "desktop",
              "value": {
                "count": 12,
                "spans": [
                  [
                    12,
                  ],
                ],
              },
            },
          ],
          "elements": [
            {
              "key": "11111111-1111-1111-1111-111111111111",
              "props": {
                "propKey": "image-test-id",
              },
              "type": "TestComponent",
            },
          ],
        },
      },
      "type": "./components/Root/index.js",
    },
    "fonts": [],
    "id": "test-page-id",
    "locale": null,
    "localizedPages": [],
    "meta": {},
    "seo": {},
    "site": {
      "id": "test-site-id",
    },
    "snippets": [],
  },
}
`;

exports[`Page Image control data v0 with format: makeswift::controls::image::format::url reading file from cache: resolvedValue 1`] = `"https://example.com/image.jpg"`;

exports[`Page Image control data v0 with format: makeswift::controls::image::format::url reading file from cache: snapshot 1`] = `
{
  "cacheData": {
    "apiResources": {
      "File": [
        {
          "id": "image-test-id",
          "value": {
            "__typename": "File",
            "dimensions": {
              "height": 100,
              "width": 100,
            },
            "extension": "jpg",
            "id": "image-test-id",
            "name": "image",
            "publicUrl": "https://example.com/image.jpg",
          },
        },
      ],
    },
    "localizedResourcesMap": {},
  },
  "document": {
    "data": {
      "key": "00000000-0000-0000-0000-000000000000",
      "props": {
        "children": {
          "columns": [
            {
              "deviceId": "desktop",
              "value": {
                "count": 12,
                "spans": [
                  [
                    12,
                  ],
                ],
              },
            },
          ],
          "elements": [
            {
              "key": "11111111-1111-1111-1111-111111111111",
              "props": {
                "propKey": "image-test-id",
              },
              "type": "TestComponent",
            },
          ],
        },
      },
      "type": "./components/Root/index.js",
    },
    "fonts": [],
    "id": "test-page-id",
    "locale": null,
    "localizedPages": [],
    "meta": {},
    "seo": {},
    "site": {
      "id": "test-site-id",
    },
    "snippets": [],
  },
}
`;

exports[`Page Image control data v0 with format: makeswift::controls::image::format::with-dimensions fetching image: resolvedValue 1`] = `
{
  "dimensions": {
    "height": 100,
    "width": 100,
  },
  "url": "https://example.com/image.jpg",
}
`;

exports[`Page Image control data v0 with format: makeswift::controls::image::format::with-dimensions fetching image: snapshot 1`] = `
{
  "cacheData": {
    "apiResources": {},
    "localizedResourcesMap": {},
  },
  "document": {
    "data": {
      "key": "00000000-0000-0000-0000-000000000000",
      "props": {
        "children": {
          "columns": [
            {
              "deviceId": "desktop",
              "value": {
                "count": 12,
                "spans": [
                  [
                    12,
                  ],
                ],
              },
            },
          ],
          "elements": [
            {
              "key": "11111111-1111-1111-1111-111111111111",
              "props": {
                "propKey": "image-test-id",
              },
              "type": "TestComponent",
            },
          ],
        },
      },
      "type": "./components/Root/index.js",
    },
    "fonts": [],
    "id": "test-page-id",
    "locale": null,
    "localizedPages": [],
    "meta": {},
    "seo": {},
    "site": {
      "id": "test-site-id",
    },
    "snippets": [],
  },
}
`;

exports[`Page Image control data v0 with format: makeswift::controls::image::format::with-dimensions reading file from cache: resolvedValue 1`] = `
{
  "dimensions": {
    "height": 100,
    "width": 100,
  },
  "url": "https://example.com/image.jpg",
}
`;

exports[`Page Image control data v0 with format: makeswift::controls::image::format::with-dimensions reading file from cache: snapshot 1`] = `
{
  "cacheData": {
    "apiResources": {
      "File": [
        {
          "id": "image-test-id",
          "value": {
            "__typename": "File",
            "dimensions": {
              "height": 100,
              "width": 100,
            },
            "extension": "jpg",
            "id": "image-test-id",
            "name": "image",
            "publicUrl": "https://example.com/image.jpg",
          },
        },
      ],
    },
    "localizedResourcesMap": {},
  },
  "document": {
    "data": {
      "key": "00000000-0000-0000-0000-000000000000",
      "props": {
        "children": {
          "columns": [
            {
              "deviceId": "desktop",
              "value": {
                "count": 12,
                "spans": [
                  [
                    12,
                  ],
                ],
              },
            },
          ],
          "elements": [
            {
              "key": "11111111-1111-1111-1111-111111111111",
              "props": {
                "propKey": "image-test-id",
              },
              "type": "TestComponent",
            },
          ],
        },
      },
      "type": "./components/Root/index.js",
    },
    "fonts": [],
    "id": "test-page-id",
    "locale": null,
    "localizedPages": [],
    "meta": {},
    "seo": {},
    "site": {
      "id": "test-site-id",
    },
    "snippets": [],
  },
}
`;

exports[`Page Image control data v0 with format: undefined fetching image: resolvedValue 1`] = `"https://example.com/image.jpg"`;

exports[`Page Image control data v0 with format: undefined fetching image: snapshot 1`] = `
{
  "cacheData": {
    "apiResources": {},
    "localizedResourcesMap": {},
  },
  "document": {
    "data": {
      "key": "00000000-0000-0000-0000-000000000000",
      "props": {
        "children": {
          "columns": [
            {
              "deviceId": "desktop",
              "value": {
                "count": 12,
                "spans": [
                  [
                    12,
                  ],
                ],
              },
            },
          ],
          "elements": [
            {
              "key": "11111111-1111-1111-1111-111111111111",
              "props": {
                "propKey": "image-test-id",
              },
              "type": "TestComponent",
            },
          ],
        },
      },
      "type": "./components/Root/index.js",
    },
    "fonts": [],
    "id": "test-page-id",
    "locale": null,
    "localizedPages": [],
    "meta": {},
    "seo": {},
    "site": {
      "id": "test-site-id",
    },
    "snippets": [],
  },
}
`;

exports[`Page Image control data v0 with format: undefined reading file from cache: resolvedValue 1`] = `"https://example.com/image.jpg"`;

exports[`Page Image control data v0 with format: undefined reading file from cache: snapshot 1`] = `
{
  "cacheData": {
    "apiResources": {
      "File": [
        {
          "id": "image-test-id",
          "value": {
            "__typename": "File",
            "dimensions": {
              "height": 100,
              "width": 100,
            },
            "extension": "jpg",
            "id": "image-test-id",
            "name": "image",
            "publicUrl": "https://example.com/image.jpg",
          },
        },
      ],
    },
    "localizedResourcesMap": {},
  },
  "document": {
    "data": {
      "key": "00000000-0000-0000-0000-000000000000",
      "props": {
        "children": {
          "columns": [
            {
              "deviceId": "desktop",
              "value": {
                "count": 12,
                "spans": [
                  [
                    12,
                  ],
                ],
              },
            },
          ],
          "elements": [
            {
              "key": "11111111-1111-1111-1111-111111111111",
              "props": {
                "propKey": "image-test-id",
              },
              "type": "TestComponent",
            },
          ],
        },
      },
      "type": "./components/Root/index.js",
    },
    "fonts": [],
    "id": "test-page-id",
    "locale": null,
    "localizedPages": [],
    "meta": {},
    "seo": {},
    "site": {
      "id": "test-site-id",
    },
    "snippets": [],
  },
}
`;

exports[`Page Image control data v1 (external image) with value: {"height": 100, "type": "external-file", "url": "https://example.com/image.jpg", "width": 100} with format: makeswift::controls::image::format::url rendering external image: resolvedValue 1`] = `"https://example.com/image.jpg"`;

exports[`Page Image control data v1 (external image) with value: {"height": 100, "type": "external-file", "url": "https://example.com/image.jpg", "width": 100} with format: makeswift::controls::image::format::url rendering external image: snapshot 1`] = `
{
  "cacheData": {
    "apiResources": {},
    "localizedResourcesMap": {},
  },
  "document": {
    "data": {
      "key": "00000000-0000-0000-0000-000000000000",
      "props": {
        "children": {
          "columns": [
            {
              "deviceId": "desktop",
              "value": {
                "count": 12,
                "spans": [
                  [
                    12,
                  ],
                ],
              },
            },
          ],
          "elements": [
            {
              "key": "11111111-1111-1111-1111-111111111111",
              "props": {
                "propKey": {
                  "height": 100,
                  "type": "external-file",
                  "url": "https://example.com/image.jpg",
                  "version": 1,
                  "width": 100,
                },
              },
              "type": "TestComponent",
            },
          ],
        },
      },
      "type": "./components/Root/index.js",
    },
    "fonts": [],
    "id": "test-page-id",
    "locale": null,
    "localizedPages": [],
    "meta": {},
    "seo": {},
    "site": {
      "id": "test-site-id",
    },
    "snippets": [],
  },
}
`;

exports[`Page Image control data v1 (external image) with value: {"height": 100, "type": "external-file", "url": "https://example.com/image.jpg", "width": 100} with format: makeswift::controls::image::format::with-dimensions rendering external image: resolvedValue 1`] = `
{
  "dimensions": {
    "height": 100,
    "width": 100,
  },
  "url": "https://example.com/image.jpg",
}
`;

exports[`Page Image control data v1 (external image) with value: {"height": 100, "type": "external-file", "url": "https://example.com/image.jpg", "width": 100} with format: makeswift::controls::image::format::with-dimensions rendering external image: snapshot 1`] = `
{
  "cacheData": {
    "apiResources": {},
    "localizedResourcesMap": {},
  },
  "document": {
    "data": {
      "key": "00000000-0000-0000-0000-000000000000",
      "props": {
        "children": {
          "columns": [
            {
              "deviceId": "desktop",
              "value": {
                "count": 12,
                "spans": [
                  [
                    12,
                  ],
                ],
              },
            },
          ],
          "elements": [
            {
              "key": "11111111-1111-1111-1111-111111111111",
              "props": {
                "propKey": {
                  "height": 100,
                  "type": "external-file",
                  "url": "https://example.com/image.jpg",
                  "version": 1,
                  "width": 100,
                },
              },
              "type": "TestComponent",
            },
          ],
        },
      },
      "type": "./components/Root/index.js",
    },
    "fonts": [],
    "id": "test-page-id",
    "locale": null,
    "localizedPages": [],
    "meta": {},
    "seo": {},
    "site": {
      "id": "test-site-id",
    },
    "snippets": [],
  },
}
`;

exports[`Page Image control data v1 (external image) with value: {"height": 100, "type": "external-file", "url": "https://example.com/image.jpg", "width": 100} with format: undefined rendering external image: resolvedValue 1`] = `"https://example.com/image.jpg"`;

exports[`Page Image control data v1 (external image) with value: {"height": 100, "type": "external-file", "url": "https://example.com/image.jpg", "width": 100} with format: undefined rendering external image: snapshot 1`] = `
{
  "cacheData": {
    "apiResources": {},
    "localizedResourcesMap": {},
  },
  "document": {
    "data": {
      "key": "00000000-0000-0000-0000-000000000000",
      "props": {
        "children": {
          "columns": [
            {
              "deviceId": "desktop",
              "value": {
                "count": 12,
                "spans": [
                  [
                    12,
                  ],
                ],
              },
            },
          ],
          "elements": [
            {
              "key": "11111111-1111-1111-1111-111111111111",
              "props": {
                "propKey": {
                  "height": 100,
                  "type": "external-file",
                  "url": "https://example.com/image.jpg",
                  "version": 1,
                  "width": 100,
                },
              },
              "type": "TestComponent",
            },
          ],
        },
      },
      "type": "./components/Root/index.js",
    },
    "fonts": [],
    "id": "test-page-id",
    "locale": null,
    "localizedPages": [],
    "meta": {},
    "seo": {},
    "site": {
      "id": "test-site-id",
    },
    "snippets": [],
  },
}
`;

exports[`Page Image control data v1 with value: {"id": "image-test-id", "type": "makeswift-file"} with format: makeswift::controls::image::format::url fetching image: resolvedValue 1`] = `"https://example.com/image.jpg"`;

exports[`Page Image control data v1 with value: {"id": "image-test-id", "type": "makeswift-file"} with format: makeswift::controls::image::format::url fetching image: snapshot 1`] = `
{
  "cacheData": {
    "apiResources": {},
    "localizedResourcesMap": {},
  },
  "document": {
    "data": {
      "key": "00000000-0000-0000-0000-000000000000",
      "props": {
        "children": {
          "columns": [
            {
              "deviceId": "desktop",
              "value": {
                "count": 12,
                "spans": [
                  [
                    12,
                  ],
                ],
              },
            },
          ],
          "elements": [
            {
              "key": "11111111-1111-1111-1111-111111111111",
              "props": {
                "propKey": {
                  "id": "image-test-id",
                  "type": "makeswift-file",
                  "version": 1,
                },
              },
              "type": "TestComponent",
            },
          ],
        },
      },
      "type": "./components/Root/index.js",
    },
    "fonts": [],
    "id": "test-page-id",
    "locale": null,
    "localizedPages": [],
    "meta": {},
    "seo": {},
    "site": {
      "id": "test-site-id",
    },
    "snippets": [],
  },
}
`;

exports[`Page Image control data v1 with value: {"id": "image-test-id", "type": "makeswift-file"} with format: makeswift::controls::image::format::url reading file from cache: resolvedValue 1`] = `"https://example.com/image.jpg"`;

exports[`Page Image control data v1 with value: {"id": "image-test-id", "type": "makeswift-file"} with format: makeswift::controls::image::format::url reading file from cache: snapshot 1`] = `
{
  "cacheData": {
    "apiResources": {
      "File": [
        {
          "id": "image-test-id",
          "value": {
            "__typename": "File",
            "dimensions": {
              "height": 100,
              "width": 100,
            },
            "extension": "jpg",
            "id": "image-test-id",
            "name": "image",
            "publicUrl": "https://example.com/image.jpg",
          },
        },
      ],
    },
    "localizedResourcesMap": {},
  },
  "document": {
    "data": {
      "key": "00000000-0000-0000-0000-000000000000",
      "props": {
        "children": {
          "columns": [
            {
              "deviceId": "desktop",
              "value": {
                "count": 12,
                "spans": [
                  [
                    12,
                  ],
                ],
              },
            },
          ],
          "elements": [
            {
              "key": "11111111-1111-1111-1111-111111111111",
              "props": {
                "propKey": {
                  "id": "image-test-id",
                  "type": "makeswift-file",
                  "version": 1,
                },
              },
              "type": "TestComponent",
            },
          ],
        },
      },
      "type": "./components/Root/index.js",
    },
    "fonts": [],
    "id": "test-page-id",
    "locale": null,
    "localizedPages": [],
    "meta": {},
    "seo": {},
    "site": {
      "id": "test-site-id",
    },
    "snippets": [],
  },
}
`;

exports[`Page Image control data v1 with value: {"id": "image-test-id", "type": "makeswift-file"} with format: makeswift::controls::image::format::with-dimensions fetching image: resolvedValue 1`] = `
{
  "dimensions": {
    "height": 100,
    "width": 100,
  },
  "url": "https://example.com/image.jpg",
}
`;

exports[`Page Image control data v1 with value: {"id": "image-test-id", "type": "makeswift-file"} with format: makeswift::controls::image::format::with-dimensions fetching image: snapshot 1`] = `
{
  "cacheData": {
    "apiResources": {},
    "localizedResourcesMap": {},
  },
  "document": {
    "data": {
      "key": "00000000-0000-0000-0000-000000000000",
      "props": {
        "children": {
          "columns": [
            {
              "deviceId": "desktop",
              "value": {
                "count": 12,
                "spans": [
                  [
                    12,
                  ],
                ],
              },
            },
          ],
          "elements": [
            {
              "key": "11111111-1111-1111-1111-111111111111",
              "props": {
                "propKey": {
                  "id": "image-test-id",
                  "type": "makeswift-file",
                  "version": 1,
                },
              },
              "type": "TestComponent",
            },
          ],
        },
      },
      "type": "./components/Root/index.js",
    },
    "fonts": [],
    "id": "test-page-id",
    "locale": null,
    "localizedPages": [],
    "meta": {},
    "seo": {},
    "site": {
      "id": "test-site-id",
    },
    "snippets": [],
  },
}
`;

exports[`Page Image control data v1 with value: {"id": "image-test-id", "type": "makeswift-file"} with format: makeswift::controls::image::format::with-dimensions reading file from cache: resolvedValue 1`] = `
{
  "dimensions": {
    "height": 100,
    "width": 100,
  },
  "url": "https://example.com/image.jpg",
}
`;

exports[`Page Image control data v1 with value: {"id": "image-test-id", "type": "makeswift-file"} with format: makeswift::controls::image::format::with-dimensions reading file from cache: snapshot 1`] = `
{
  "cacheData": {
    "apiResources": {
      "File": [
        {
          "id": "image-test-id",
          "value": {
            "__typename": "File",
            "dimensions": {
              "height": 100,
              "width": 100,
            },
            "extension": "jpg",
            "id": "image-test-id",
            "name": "image",
            "publicUrl": "https://example.com/image.jpg",
          },
        },
      ],
    },
    "localizedResourcesMap": {},
  },
  "document": {
    "data": {
      "key": "00000000-0000-0000-0000-000000000000",
      "props": {
        "children": {
          "columns": [
            {
              "deviceId": "desktop",
              "value": {
                "count": 12,
                "spans": [
                  [
                    12,
                  ],
                ],
              },
            },
          ],
          "elements": [
            {
              "key": "11111111-1111-1111-1111-111111111111",
              "props": {
                "propKey": {
                  "id": "image-test-id",
                  "type": "makeswift-file",
                  "version": 1,
                },
              },
              "type": "TestComponent",
            },
          ],
        },
      },
      "type": "./components/Root/index.js",
    },
    "fonts": [],
    "id": "test-page-id",
    "locale": null,
    "localizedPages": [],
    "meta": {},
    "seo": {},
    "site": {
      "id": "test-site-id",
    },
    "snippets": [],
  },
}
`;

exports[`Page Image control data v1 with value: {"id": "image-test-id", "type": "makeswift-file"} with format: undefined fetching image: resolvedValue 1`] = `"https://example.com/image.jpg"`;

exports[`Page Image control data v1 with value: {"id": "image-test-id", "type": "makeswift-file"} with format: undefined fetching image: snapshot 1`] = `
{
  "cacheData": {
    "apiResources": {},
    "localizedResourcesMap": {},
  },
  "document": {
    "data": {
      "key": "00000000-0000-0000-0000-000000000000",
      "props": {
        "children": {
          "columns": [
            {
              "deviceId": "desktop",
              "value": {
                "count": 12,
                "spans": [
                  [
                    12,
                  ],
                ],
              },
            },
          ],
          "elements": [
            {
              "key": "11111111-1111-1111-1111-111111111111",
              "props": {
                "propKey": {
                  "id": "image-test-id",
                  "type": "makeswift-file",
                  "version": 1,
                },
              },
              "type": "TestComponent",
            },
          ],
        },
      },
      "type": "./components/Root/index.js",
    },
    "fonts": [],
    "id": "test-page-id",
    "locale": null,
    "localizedPages": [],
    "meta": {},
    "seo": {},
    "site": {
      "id": "test-site-id",
    },
    "snippets": [],
  },
}
`;

exports[`Page Image control data v1 with value: {"id": "image-test-id", "type": "makeswift-file"} with format: undefined reading file from cache: resolvedValue 1`] = `"https://example.com/image.jpg"`;

exports[`Page Image control data v1 with value: {"id": "image-test-id", "type": "makeswift-file"} with format: undefined reading file from cache: snapshot 1`] = `
{
  "cacheData": {
    "apiResources": {
      "File": [
        {
          "id": "image-test-id",
          "value": {
            "__typename": "File",
            "dimensions": {
              "height": 100,
              "width": 100,
            },
            "extension": "jpg",
            "id": "image-test-id",
            "name": "image",
            "publicUrl": "https://example.com/image.jpg",
          },
        },
      ],
    },
    "localizedResourcesMap": {},
  },
  "document": {
    "data": {
      "key": "00000000-0000-0000-0000-000000000000",
      "props": {
        "children": {
          "columns": [
            {
              "deviceId": "desktop",
              "value": {
                "count": 12,
                "spans": [
                  [
                    12,
                  ],
                ],
              },
            },
          ],
          "elements": [
            {
              "key": "11111111-1111-1111-1111-111111111111",
              "props": {
                "propKey": {
                  "id": "image-test-id",
                  "type": "makeswift-file",
                  "version": 1,
                },
              },
              "type": "TestComponent",
            },
          ],
        },
      },
      "type": "./components/Root/index.js",
    },
    "fonts": [],
    "id": "test-page-id",
    "locale": null,
    "localizedPages": [],
    "meta": {},
    "seo": {},
    "site": {
      "id": "test-site-id",
    },
    "snippets": [],
  },
}
`;

exports[`Page ImageDefinition v1 reading v0 data with format: makeswift::controls::image::format::url fetching image: resolvedValue 1`] = `"https://example.com/image.jpg"`;

exports[`Page ImageDefinition v1 reading v0 data with format: makeswift::controls::image::format::url fetching image: snapshot 1`] = `
{
  "cacheData": {
    "apiResources": {},
    "localizedResourcesMap": {},
  },
  "document": {
    "data": {
      "key": "00000000-0000-0000-0000-000000000000",
      "props": {
        "children": {
          "columns": [
            {
              "deviceId": "desktop",
              "value": {
                "count": 12,
                "spans": [
                  [
                    12,
                  ],
                ],
              },
            },
          ],
          "elements": [
            {
              "key": "11111111-1111-1111-1111-111111111111",
              "props": {
                "propKey": {
                  "id": "image-test-id",
                  "type": "makeswift-file",
                  "version": 1,
                },
              },
              "type": "TestComponent",
            },
          ],
        },
      },
      "type": "./components/Root/index.js",
    },
    "fonts": [],
    "id": "test-page-id",
    "locale": null,
    "localizedPages": [],
    "meta": {},
    "seo": {},
    "site": {
      "id": "test-site-id",
    },
    "snippets": [],
  },
}
`;

exports[`Page ImageDefinition v1 reading v0 data with format: makeswift::controls::image::format::url reading file from cache: resolvedValue 1`] = `"https://example.com/image.jpg"`;

exports[`Page ImageDefinition v1 reading v0 data with format: makeswift::controls::image::format::url reading file from cache: snapshot 1`] = `
{
  "cacheData": {
    "apiResources": {
      "File": [
        {
          "id": "image-test-id",
          "value": {
            "__typename": "File",
            "dimensions": {
              "height": 100,
              "width": 100,
            },
            "extension": "jpg",
            "id": "image-test-id",
            "name": "image",
            "publicUrl": "https://example.com/image.jpg",
          },
        },
      ],
    },
    "localizedResourcesMap": {},
  },
  "document": {
    "data": {
      "key": "00000000-0000-0000-0000-000000000000",
      "props": {
        "children": {
          "columns": [
            {
              "deviceId": "desktop",
              "value": {
                "count": 12,
                "spans": [
                  [
                    12,
                  ],
                ],
              },
            },
          ],
          "elements": [
            {
              "key": "11111111-1111-1111-1111-111111111111",
              "props": {
                "propKey": {
                  "id": "image-test-id",
                  "type": "makeswift-file",
                  "version": 1,
                },
              },
              "type": "TestComponent",
            },
          ],
        },
      },
      "type": "./components/Root/index.js",
    },
    "fonts": [],
    "id": "test-page-id",
    "locale": null,
    "localizedPages": [],
    "meta": {},
    "seo": {},
    "site": {
      "id": "test-site-id",
    },
    "snippets": [],
  },
}
`;

exports[`Page ImageDefinition v1 reading v0 data with format: makeswift::controls::image::format::with-dimensions fetching image: resolvedValue 1`] = `
{
  "dimensions": {
    "height": 100,
    "width": 100,
  },
  "url": "https://example.com/image.jpg",
}
`;

exports[`Page ImageDefinition v1 reading v0 data with format: makeswift::controls::image::format::with-dimensions fetching image: snapshot 1`] = `
{
  "cacheData": {
    "apiResources": {},
    "localizedResourcesMap": {},
  },
  "document": {
    "data": {
      "key": "00000000-0000-0000-0000-000000000000",
      "props": {
        "children": {
          "columns": [
            {
              "deviceId": "desktop",
              "value": {
                "count": 12,
                "spans": [
                  [
                    12,
                  ],
                ],
              },
            },
          ],
          "elements": [
            {
              "key": "11111111-1111-1111-1111-111111111111",
              "props": {
                "propKey": {
                  "id": "image-test-id",
                  "type": "makeswift-file",
                  "version": 1,
                },
              },
              "type": "TestComponent",
            },
          ],
        },
      },
      "type": "./components/Root/index.js",
    },
    "fonts": [],
    "id": "test-page-id",
    "locale": null,
    "localizedPages": [],
    "meta": {},
    "seo": {},
    "site": {
      "id": "test-site-id",
    },
    "snippets": [],
  },
}
`;

exports[`Page ImageDefinition v1 reading v0 data with format: makeswift::controls::image::format::with-dimensions reading file from cache: resolvedValue 1`] = `
{
  "dimensions": {
    "height": 100,
    "width": 100,
  },
  "url": "https://example.com/image.jpg",
}
`;

exports[`Page ImageDefinition v1 reading v0 data with format: makeswift::controls::image::format::with-dimensions reading file from cache: snapshot 1`] = `
{
  "cacheData": {
    "apiResources": {
      "File": [
        {
          "id": "image-test-id",
          "value": {
            "__typename": "File",
            "dimensions": {
              "height": 100,
              "width": 100,
            },
            "extension": "jpg",
            "id": "image-test-id",
            "name": "image",
            "publicUrl": "https://example.com/image.jpg",
          },
        },
      ],
    },
    "localizedResourcesMap": {},
  },
  "document": {
    "data": {
      "key": "00000000-0000-0000-0000-000000000000",
      "props": {
        "children": {
          "columns": [
            {
              "deviceId": "desktop",
              "value": {
                "count": 12,
                "spans": [
                  [
                    12,
                  ],
                ],
              },
            },
          ],
          "elements": [
            {
              "key": "11111111-1111-1111-1111-111111111111",
              "props": {
                "propKey": {
                  "id": "image-test-id",
                  "type": "makeswift-file",
                  "version": 1,
                },
              },
              "type": "TestComponent",
            },
          ],
        },
      },
      "type": "./components/Root/index.js",
    },
    "fonts": [],
    "id": "test-page-id",
    "locale": null,
    "localizedPages": [],
    "meta": {},
    "seo": {},
    "site": {
      "id": "test-site-id",
    },
    "snippets": [],
  },
}
`;

exports[`Page ImageDefinition v1 reading v0 data with format: undefined fetching image: resolvedValue 1`] = `"https://example.com/image.jpg"`;

exports[`Page ImageDefinition v1 reading v0 data with format: undefined fetching image: snapshot 1`] = `
{
  "cacheData": {
    "apiResources": {},
    "localizedResourcesMap": {},
  },
  "document": {
    "data": {
      "key": "00000000-0000-0000-0000-000000000000",
      "props": {
        "children": {
          "columns": [
            {
              "deviceId": "desktop",
              "value": {
                "count": 12,
                "spans": [
                  [
                    12,
                  ],
                ],
              },
            },
          ],
          "elements": [
            {
              "key": "11111111-1111-1111-1111-111111111111",
              "props": {
                "propKey": {
                  "id": "image-test-id",
                  "type": "makeswift-file",
                  "version": 1,
                },
              },
              "type": "TestComponent",
            },
          ],
        },
      },
      "type": "./components/Root/index.js",
    },
    "fonts": [],
    "id": "test-page-id",
    "locale": null,
    "localizedPages": [],
    "meta": {},
    "seo": {},
    "site": {
      "id": "test-site-id",
    },
    "snippets": [],
  },
}
`;

exports[`Page ImageDefinition v1 reading v0 data with format: undefined reading file from cache: resolvedValue 1`] = `"https://example.com/image.jpg"`;

exports[`Page ImageDefinition v1 reading v0 data with format: undefined reading file from cache: snapshot 1`] = `
{
  "cacheData": {
    "apiResources": {
      "File": [
        {
          "id": "image-test-id",
          "value": {
            "__typename": "File",
            "dimensions": {
              "height": 100,
              "width": 100,
            },
            "extension": "jpg",
            "id": "image-test-id",
            "name": "image",
            "publicUrl": "https://example.com/image.jpg",
          },
        },
      ],
    },
    "localizedResourcesMap": {},
  },
  "document": {
    "data": {
      "key": "00000000-0000-0000-0000-000000000000",
      "props": {
        "children": {
          "columns": [
            {
              "deviceId": "desktop",
              "value": {
                "count": 12,
                "spans": [
                  [
                    12,
                  ],
                ],
              },
            },
          ],
          "elements": [
            {
              "key": "11111111-1111-1111-1111-111111111111",
              "props": {
                "propKey": {
                  "id": "image-test-id",
                  "type": "makeswift-file",
                  "version": 1,
                },
              },
              "type": "TestComponent",
            },
          ],
        },
      },
      "type": "./components/Root/index.js",
    },
    "fonts": [],
    "id": "test-page-id",
    "locale": null,
    "localizedPages": [],
    "meta": {},
    "seo": {},
    "site": {
      "id": "test-site-id",
    },
    "snippets": [],
  },
}
`;
